downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -39,6 +41,18 @@ open class GitWorkingTree(workingDir: File, private val gitBase: GitBase) : Work
         return File(dotGitDir, "shallow").isFile
     }
 
+    override fun getNested(): Map<String, VcsInfo> {
+        val root = getRootPath()
+
+        val paths = gitBase.run(root, "submodule", "--quiet", "foreach", "--recursive", "echo \$sm_path").stdout.lines()
+                .filter { it.isNotBlank() }
```

Comment:
Is this really needed? So far I never saw blank lines in the output.

Reply:
The last line is always blank, but it seemed safer to me to check all lines.